src/Hipace.cpp

```diff
@@ -1506,13 +1517,20 @@ Hipace::Notify (const int step, const int it,
         const std::size_t nreals = lbx.numPts()*laser_fab.nComp();
         m_lsend_buffer = (amrex::Real*)amrex::The_Pinned_Arena()->alloc
             (sizeof(amrex::Real)*nreals);
-        auto const buf = amrex::makeArray4(m_lsend_buffer, lbx, laser_fab.nComp());
-        amrex::ParallelFor
-            (lbx, laser_fab.nComp(), [=] AMREX_GPU_DEVICE (int i, int j, int k, int n) noexcept
-            {
-                buf(i,j,k,n) = laser_arr(i,j,k,n);
-            });
-        amrex::Gpu::Device::synchronize();
+        if (m_laser.is3dOnHost()) {
+            // Copy from laser envelope 3D array (on host) to MPI buffer (on host)
```

Comment:
There should be another stream synchronize before the copy to mem to make sure that laser_fab is fully populated by the GPU. Or do you know of a different place between here and where values are copied into laser_fab that is guaranteed to synchronize?

Reply:
https://github.com/Hi-PACE/hipace/blob/b14edb0a5ca9e5ab24f7c597c6a1ce52857966a2/src/laser/Laser.cpp#L156-L213